modules/nlu/src/backend/election/natural-election.ts

```diff
@@ -45,20 +48,47 @@ function electIntent(input: sdk.IO.EventUnderstanding): sdk.IO.EventUnderstandin
   }
 }
 
-function extractElectedIntentSlot(input: sdk.IO.EventUnderstanding): sdk.IO.EventUnderstanding {
-  if (!input.predictions) {
+export function extractElectedIntentSlot(input: sdk.IO.EventUnderstanding): sdk.IO.EventUnderstanding {
+  if (!isNotNil(input.predictions)) {
```

Comment:
The double negation is weird tho: `!isNot`

Reply:
Nil is Null/Undefined.

The function is a utilitarian one that can serve us throughout the project. It's stronger than `!someVariable` because it strictly checks on `null` and `undefined` values instead of all falsy values, which can include values like `0`, `false`, etc.

Furthermore, it's a handy function for filtering an array, leaving only non-nil values in, [like such](https://www.typescriptlang.org/play?#code/GYVwdgxgLglg9mABDAzgOTlNMA2AeAFQD4AKADwC4CAfcAEwFNgYwG7qwQccBKCs5CkQFEAbwBQiKYgBODKCBlIBAQgC8axJ26IAZLsSqNiekxZtxAX3HiICFFEQBDGTMSaA2tpwAaRAEY-ACY-AGY-ABY-bwBdADpmHCgGGRJUDCxcHgBuW3s4HAY4nDgAcxIXGR4gA):
```typescript
const arr = [null, 1, 2, 3, 4, null].filter(isNotNil);
console.log(arr)
// [1, 2, 3, 4]
```